File: Problem18/euler.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    int t, a[15][15];
    cin>>t;
    while(t--)
    {
        int n;
        cin>>n;
        for(int i = 0; i < n; i++)
            for(int j = 0; j <= i; j++)
                cin>>a[i][j];
        int sum = 0;
        for (int i = 0; i < n; i ++)
            for(int j = 0; j <= i; j++)
        {
            int pathleft = j == 0 ? 0 : a[i-1][j-1];
            int pathright = j == i ? 0 : a[i-1][j];
            a[i][j] += pathleft > pathright ? pathleft : pathright;  
        }
        int max = 0;
        for (int j =0; j < n; j++)
            if(max < a[n-1][j])
                max = a[n-1][j];
        cout<<max<<endl;
    }
    return 0;
}

File: Problem67/euler.py
for t in range(int(raw_input())):
    n = int(raw_input())
    array = []
    for i in range(n):
        array.append(map(int, raw_input().split(" ")))
    for i in range(n):
        for j in range(i+1):
            if j == 0:
                pathleft = 0
            else:
                pathleft = array[i-1][j-1]
            if j == i:
                pathright = 0
            else:
                pathright = array[i-1][j]
            array[i][j] = array[i][j] + max(pathleft, pathright)
    print max(array[n-1])